presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -629,6 +632,26 @@ public static Slice stEnvelope(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return serialize(envelope);
     }
 
+    @Description("Returns the lower left and upper right corners of bounding rectangular polygon of a Geometry")
+    @ScalarFunction("ST_EnvelopeAsPts")
+    @SqlType("array(" + GEOMETRY_TYPE_NAME + ")")
+    public static Block stEnvelopeAsPts(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        Envelope envelope = deserializeEnvelope(input);
+        BlockBuilder blockBuilder = VARCHAR.createBlockBuilder(null, 2);
```

Comment:
Why VARCHAR? Shouldn't it be a GEOMETRY_TYPE?

Reply:
Done